File: sketches/date.rb
date_file = "date.txt"
system "date > #{date_file}"

fp = open date_file, "r"
string = fp.read
words = string.split
time = nil
words.each do |word|
  time = word if word[/\d+:\d+:\d+/]
end
time = time.split(":")
time[0] = String(Integer(time[0]) + 3)
if Integer(time[0]) < 10
  time[0] = time[0].concat("0").reverse
elsif Integer(time[0]) > 24
  time[0] = String(Integer(time[0]) - 24)
end
new_hour = ""
time.each do |hour|
  new_hour.concat(hour)
  new_hour.concat(":")
end
new_hour.chomp!(":")
new_date = ""
words.each do |word|
  if word[/\d+:\d+:\d+/]
    new_date.concat(new_hour)
  else
    new_date.concat(word)
  end
  new_date.concat(" ")
end
new_date.chomp!

# puts new_date
system "sudo date --set=\"#{new_date}\""
system "rm #{date_file}"

File: work/date.py
import os
import re

# creation of words
os.system('date > date.txt')
fp = open('date.txt', 'r')
words = fp.read().split()
fp.close()

# remotion of the hour part
hour = ''
for word in words:
  mo = re.search(r'\d+:\d+:\d+', word)
  if mo: hour = mo.group()

# change of name
hour = hour.split(':')
hour[0] = int(hour[0]) + 3
if hour[0] > 24:
  hour[0] = hour[0] - 24
hour[0] = str(hour[0])
new_hour = ''
for h in hour:
  new_hour = '{}{}:'.format(new_hour, h)
new_hour = new_hour.rstrip(':')

# construction of new hour
new_date = ''
for word in words:
  mo = re.search(r'\d+:\d+:\d+', word)
  if mo:
    new_date = '{} {}'.format(new_date, new_hour)
  else:
    new_date = '{} {}'.format(new_date, word)

# set up of new date
# print new_date.strip()
os.system('sudo date --set=\'{}\''.format(new_date))
